### DISCUSSION TITLE
Integrating MicroPython ADF and Camera API on ESP32-S3: Creating a Unified Firmware

### DISCUSSION THREAD

**Discussion by sheetaljain01** [micropython/micropython#16273] [Core Development]
Hi,  

I am currently working on an ESP32-S3 camera with esp-adf on micropython.  

I have tested both of the following:  
1. Micropython ESP32-S3 camera with this repository:  
   [https://github.com/cnadler86/micropython-camera-API](https://github.com/cnadler86/micropython-camera-API)  

2. MicroPython ADF with this repository:  
   [https://github.com/espressif/esp-adf/tree/master/micropython_adf](https://github.com/espressif/esp-adf/tree/master/micropython_adf)  

When I create custom firmware, both use different build commands:  

- For the camera, I use:  

          make USER_C_MODULES=../../../../micropython-camera-API/src/micropython.cmake BOARD=ESP32_GENERIC_S3 
          BOARD_VARIANT=SPIRAM_OCT all  
 

- For MicroPython ADF, I use:  

        idf.py build -D MICROPY_BOARD_DIR=${ADF_PATH}/micropython_adf/boards/korvo2v3 -D 
        USER_C_MODULES=${ADF_PATH}/micropython_adf/mod/micropython.cmake  

How can I merge both of them to create a single firmware file where both functionalities work together?

**Comment by bidrohini** [comment]
Both repositories (micropython-camera-API and micropython_adf) have custom modules and potentially overlapping configurations. You have to identify the modules or features they add and any conflicting dependencies or settings.

**Comment by jonnor** [comment]
Where you able to make any progress on this @sheetaljain01 ?

To combine different user C modules, I believe you need to add your own .cmake file which has the contents of both of the two cmake files. And then pass that to make using USER_C_MODULES
